feedback/feedback.py

```diff
@@ -102,141 +117,81 @@ def resource_string(cls, path):
-            icon_file = template.format(i=i, set=prompt['icon_set'])
+            """
+            icon_file = icon_path_template[icon_type].format(i=icon_number, set=self.prompt["icon_set"])
             return self.runtime.local_resource_url(self, icon_file)
-        ina_urls = [get_url('inactive', i) for i in range(1, 6)]
-        act_urls = [get_url('active', i) for i in range(1, 6)]
-        sel_urls = [get_url('selected', i) for i in range(1, 6)]
-
-        # Render the Likert scale (not the whole page)
-        scale = u"".join(
-            scale_item.format(scale_text=scale_text,
-                              unicode_icon=unicode_icon,
-                              idx=idx,
-                              active=active,
-                              vote_cnt=vote_cnt,
-                              ina_icon=ina_icon,
-                              act_icon=act_icon,
-                              sel_icon=sel_icon) for
-            (scale_text,
-             unicode_icon,
-             idx,
-             active,
-             vote_cnt,
-             act_icon,
-             ina_icon,
-             sel_icon) in
-            zip(prompt['scale_text'],
-                ICON_SETS[(prompt['icon_set'])],
-                indexes,
-                active_vote,
-                votes,
-                act_urls,
-                ina_urls,
-                sel_urls)
-        )
+
+        ina_urls = [get_url("inactive", i) for i in range(1, propts_size)]
+        act_urls = [get_url("active", i) for i in range(1, propts_size)]
+
+        _ = self.runtime.service(self, "i18n").ugettext
         if self.user_vote != -1:
-            _ = self.runtime.service(self, 'i18n').ugettext
+
             response = _("Thank you for voting!")
         else:
             response = ""
-        # Now, render the whole page
-        rendered = html.format(self=self,
-                               scale=scale,
-                               freeform_prompt=prompt['freeform'],
-                               likert_prompt=prompt['likert'],
-                               response=response,
-                               placeholder=prompt['placeholder'])
+
+        zero_likert_item = [self.prompt["zero_likert"], get_url("inactive", 0), get_url("active", 0)]
+
+        context = {
+            "user_freeform": self.user_freeform,
+            "scale": self.prompt["scale_text"],
+            "ina_urls": ina_urls,
+            "act_urls": act_urls,
+            "zero_likert_item": zero_likert_item,
+            "freeform_prompts": _(self.prompt["freeform"]),
+            "likert_prompts": _(self.prompt["likert"]),
+            "response": response,
+            "placeholder": _(self.prompt["placeholder"]),
+            "enable_text_area_answer": self.enable_text_area_answer,
+            "votes": votes,
+            "active_vote": active_vote,
+            "enable_zero_grade": self.enable_zero_grade,
+            "submit": _("Submit Feedback")
```

Comment:
Why we do it here? Maybe the more convenient way is to move this text to the template?

Reply:
I moved it here, because I can't translate it another way